### PR TITLE
code commit for log based size rate limit

### PR DESCRIPTION
**Pull request by Avleen-Kathuria**

### CODE REVIEW

File: data-sdk/src/main/java/com/logicmonitor/sdk/data/api/Logs.java [hunk 1]
```diff
@@ -144,6 +162,11 @@ private List<Map<String, Object>> createBody() {
         body.put("metadata", logsV1.getMetadata());
       }
       logBody.add(body);
+      double msg_size =
+          Double.parseDouble(df.format(logsV1.getMessage().getBytes().length / 1024.0));
+      if (msg_size > Constant.DEFAULT_PUSHMETRICS_MAXIMUM_ARRAY_SIZE_FOR_SINGLE_LOG_MESSAGE) {
+        log.info("Your message exceeds 32KB It will be truncate");
```
Comment (mukundneharkar) [inline data-sdk/src/main/java/com/logicmonitor/sdk/data/api/Logs.java:168]: Should be warning..

**Review by mukundneharkar — COMMENTED** [review]
minor comments

File: data-sdk/src/main/java/com/logicmonitor/sdk/data/api/Logs.java [hunk 2]
```diff
@@ -75,6 +80,10 @@ protected static ApiResponse singleRequest(final LogsInput logsV1)
       body.put("metadata", logsV1.getMetadata());
     }
     logBody.add(body);
+    double msg_size = Double.parseDouble(df.format(logsV1.getMessage().getBytes().length / 1024.0));
+    if (msg_size > Constant.DEFAULT_PUSHMETRICS_MAXIMUM_ARRAY_SIZE_FOR_SINGLE_LOG_MESSAGE) {
+      log.info("Your message exceeds 32KB It will be truncate");
```
[the remark this answers is not in the document]
Comment (Avleen-Kathuria) [reply data-sdk/src/main/java/com/logicmonitor/sdk/data/api/Logs.java]: done the needful

File: data-sdk/src/main/java/com/logicmonitor/sdk/data/api/Logs.java (hunk 1 above)
Comment (Avleen-Kathuria) [reply data-sdk/src/main/java/com/logicmonitor/sdk/data/api/Logs.java]: done the needful

**Review by Gaurang-Patel — APPROVED** [review]

**Review by mukundneharkar — APPROVED** [review]

**Pull request merged**